src/index.js

```diff
@@ -177,6 +177,14 @@ class Node extends EventEmitter {
       }
     })
 
+    // detect which multiaddrs we don't have a transport for and remove them
+    multiaddrs.forEach((multiaddr) => {
+      if (!multiaddr.toString().match(/\/p2p-circuit($|\/)/) &&
```

Comment:
@diasdavid I'm not particularly proud of this line of code, but I learned that if I remove these addresses (although they apparently have no corresponding transport), tests start failing.. Any insight?

Reply:
Why not use https://github.com/whyrusleeping/js-mafmt ?

Reply:
@diasdavid this is not validation, the goal of this is to remove the multiaddrs for which the node has no transport for..

Reply:
(see next LOC)

Reply:
Can't you use the filter from each transport to then identify the subset of multiaddrs to which no transport has grabbed?

Reply:
I'd have to create a separate set named "used multiaddresses" to which I would add the selected multiaddresses for each transport. After that I would have to compare the "used multiaddresses" set to the current multiaddress set and remove all those that are not present.

(and the line we're discussing is not about the filtering , that's the next line. This line is about not removing multiaddresses that have `/p2p-circuit` in them..)

Reply:
Yes, and that should happen after the swarm.listen step so that multiaddrs such that have things like `0.0.0.0` and port `0` get updated first. This would be a good test to add as well

Reply:
@pgte did you get this last comment?

Reply:
Yup, will be working on this 👍

Reply:
@diasdavid please check 67db4e2060a8081be29f6b5e38511b40802ef177, where the final set of multiaddrs is being computed *after* start.

If it's really important I could use sets to do this math, but I'd have to do it very explicitly as currently MultiAddrSet (https://github.com/libp2p/js-peer-info/blob/master/src/multiaddr-set.js) does not seam to support the necessary set operations..